### PR TITLE
`Development`: Delete old files properly when updating file submissions

### PR DESCRIPTION
**Pull request by MarkusPaulsen**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context
The parameter fileUploadSubmission is currently getting used for getting the last submission in FileUploadSubmissionService. The data of this parameter is under the control of the user, which may pose certain risks.

### Description
Instead of FileUploadSubmission fileUploadSubmission, we now use StudentParticipation participation for getting the last submission in FileUploadSubmissionService, as the data of this parameter is not under the control of the user.

### Steps for Testing
We just want to make sure that I didn't accidentally break the implementation.

Prerequisites:
- 1 Instructor
- 1 Student

Fileupload Exercise (Instructor, Student):
1. Create a file upload exercise with a description. Upload an image to the description
2. Start the exercise, the image should show up properly as well as the rest of the description
3. Participate in the exercise
4. Upload another file
5. View/Download the image
6. No issues should occur

#### Performance Review
- [x] I confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by MaximilianAnzinger — DISMISSED** [review]
Code LGTM 👍

**Review by SolizerCodes — DISMISSED** [review]
Tested on ts6, works well 👍

**Review by undernagruzez — DISMISSED** [review]
Tested on ts6, works as expected

**Review by RY997 — APPROVED** [review]
Code LGTM

**Review by dearjasmina — DISMISSED** [review]
tested on ts 3, works as expected, code looks good

**Review by bassner — CHANGES_REQUESTED** [review]
In my opinion we should test these changes as they are relevant for the security of the application.

**Review by Strohgelaender — CHANGES_REQUESTED** [review]
The server style action is currently failing, please run `spotlessApply` to fix the reported issues.

File: src/main/java/de/tum/in/www1/artemis/service/FileUploadSubmissionService.java [hunk 1]
```diff
@@ -139,39 +139,42 @@ public FileUploadSubmission save(FileUploadSubmission fileUploadSubmission, Mult
...
+            .filter(previousSubmission -> previousSubmission.getFilePath() != null)
+            .ifPresent(previousSubmission -> {
+                final String oldFilePath = previousSubmission.getFilePath();
+                // check if we already had a file associated with this submission
+                if (!oldFilePath.equals(newFilePath)) { // different name
+                    // IMPORTANT: only delete the file when it has changed the name
+                    previousSubmission.onDelete();
+                } else { // same name
+                    // IMPORTANT: invalidate the cache so that the new file with the same name will be downloaded (and not a potentially cached one)
+                    fileService.resetOnPath(savePath);
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/service/FileUploadSubmissionService.java:175]: Maybe change the method name to something like `evictCacheForPath`? By just reading the code I would have not thought that this method is used for caching-related functionality.

**Review by RY997 — APPROVED** [review]
Reapprove

**Review by krusche — APPROVED** [review]
Code looks good to me and makes sense 👍

**Review by dearjasmina — APPROVED** [review]
reapprove

**Review by Strohgelaender — APPROVED** [review]
code looks good.

**Pull request merged**